.github/workflows/status-checks.yml

```diff
@@ -75,6 +75,12 @@ jobs:
     - name: Run misspell
       run: misspell -error .
 
+    - name: Install staticcheck
+      run: go install honnef.co/go/tools/cmd/staticcheck@2023.1.3
+
+    - name: Run staticcheck
+      run: staticcheck -checks "all,-ST1000,-ST1003,-SA1019,-SA1029" ./...
```

Comment:
[ST1003: Poorly chosen identifier](https://staticcheck.dev/docs/checks/#ST1003):

We currently don't follow these conventions, and if we would then we would have to rename several functions and fields in the public API. I didn't want to make that much breaking changes. Also, one of the package names in the internal package (disable_sigpipe) is not correct, so we would have to rename this package too.

[SA1029 - Inappropriate key in call to context.WithValue](https://staticcheck.dev/docs/checks/#SA1029)

We also currently do this and I did not do the effort to refactor this out. We can still do so, but then I'd prefer to do that in a separate PR.

Reply:
ST1003: Noticed disable_sigpipe myself too when checking, but agree keeping it as is to prevent breaking changes.

SA1029: I've created a separate task on the backlog for SA1029 so we won't forget about it